### ISSUE TITLE
Google Benchmark does not work out of the box

### ISSUE THREAD

**Issue by Voxel7** [microsoft/vcpkg#7654]
**Describe the bug**
The Google Benchmark can be built and exported as a NuGet package without problem. However, the package cannot be used on a MSVC project _as is_: a dependency to `shlwapi` is missing, as described in the library instruction:
https://github.com/google/benchmark#platform-specific-instructions

`benchmark.lib(sysinfo.cc.obj) : error LNK2001: unresolved external symbol __imp__SHGetValueA@24`

I know I can manually add a dependency in my project to fix it, that's not the point. This is the responsibility of a package to specify its dependencies, not the responsibility of the user.

**Environment**
- OS: Windows 10, 
- Compiler: Visual Studio 2015

**To Reproduce**
1. `./vcpkg install benchmark`
2. `./vcpkg export benchmark --nuget --nuget-id=Google.Benchmark --nuget-version=1.5.0`
3. Install the package in a Visual Studio C++ project
4. Try to build it
It should failed: `benchmark.lib(sysinfo.cc.obj) : error LNK2001: unresolved external symbol __imp__SHGetValueA@24`

**Expected behavior**
The build should succeed

**Comment by JackBoosY** [comment]
Compressed packages imported using nuget instead of cmake should not have library dependencies that come with the OS. @vicroms Can you confirm it?

**Comment by JackBoosY** [comment]
Related: #8864.

**Comment by JackBoosY** [comment]
cc @BillyONeal

**Comment by BillyONeal** [comment]
> Compressed packages imported using nuget instead of cmake should not have library dependencies that come with the OS. @vicroms Can you confirm it?

This is correct. If you are using cmake we can export that dependency in the resulting target, but with a static lib I don't think there's anything vcpkg can do.

**Comment by BillyONeal** [comment]
@Voxel7 Do you have a suggestion on what we should do?

**Comment by JonLiu1993** [comment]
Issue still exists
````
Build started...
1>------ Build started: Project: ConsoleApplication18, Configuration: Debug Win32 ------
1>ConsoleApplication18.cpp
1>benchmark.lib(sysinfo.cc.obj) : error LNK2019: unresolved external symbol __imp__SHGetValueA@24 referenced in function "double __cdecl benchmark::`anonymous namespace'::GetCPUCyclesPerSecond(enum benchmark::CPUInfo::Scaling)" (?GetCPUCyclesPerSecond@?A0xccba857a@benchmark@@YANW4Scaling@CPUInfo@2@@Z)
1>C:\Users\source\repos\ConsoleApplication18\Debug\ConsoleApplication18.exe : fatal error LNK1120: 1 unresolved externals
1>Done building project "ConsoleApplication18.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
````

**Comment by vbaderks** [comment]
@JonLiu1993 
There was  an open issue for it in 2016: https://github.com/google/benchmark/issues/202
It could be resolved with: #pragma comment(lib, "shlwapi.lib"), but only for MSVC++ users (and probably Clang) but not for GCC (MinGW). Nobody provided a patch for it, so it was closed by updating the Benchmark docs that this shlwapi.lib is needed.

**Comment by spectre-ns** [comment]
This is super annoying.

**Comment by dg0yt** [comment]
Well, vcpkg can do the following:

> It could be resolved with: #pragma comment(lib, "shlwapi.lib"), but only for MSVC++ users (and probably Clang) but not for GCC (MinGW). 

Vcpkg already is doing similar, cf. `git grep "pragma comment"  ports/`

**Comment by PhoebeHui** [comment]
@dg0yt, sure,we could patch Benchmark for msvc++ users in vcpkg, but I'm not sure why the Upstream doesn't accept the changes, and only added it to a doc. We will open a PR and see what's the suggestions from others.

**Comment by dg0yt** [comment]
> I'm not sure why the Upstream doesn't accept the changes, and only added it to a doc

Because they don't care so much about MSVC, and nobody submitted a PR?
